### PR TITLE
psp-8018: display retired warning immediately upon add.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: source/frontend/src/components/propertySelector/search/PropertySelectorSearchContainer.tsx [hunk 1]
```diff
@@ -87,17 +91,28 @@ export const PropertySelectorSearchContainer: React.FC<IPropertySelectorSearchCo
...
-              STREET_ADDRESS_1: addressResults[i]?.fullAddress,
-            },
-          } as Feature<Geometry, PIMS_Property_Location_View>;
+          foundProperty.pimsFeature = pimsResults[i]?.features?.length
+            ? pimsResults[i]?.features[0]
+            : ({
+                properties: {
+                  STREET_ADDRESS_1: addressResults[i]?.fullAddress,
+                },
+              } as Feature<Geometry, PIMS_Property_Location_View>);
```
Comment (FuriousLlama) [inline source/frontend/src/components/propertySelector/search/PropertySelectorSearchContainer.tsx:115]: This casting is smelly

**Review by FuriousLlama — APPROVED** [review]

File: source/frontend/src/components/propertySelector/search/PropertySelectorSearchContainer.tsx (hunk 1 above)
Comment (devinleighsmith) [reply source/frontend/src/components/propertySelector/search/PropertySelectorSearchContainer.tsx:115]: This isn't one of my changes, so I'll leave it as-is.

File: source/frontend/src/components/propertySelector/MapSelectorContainer.tsx [hunk 2]
```diff
@@ -55,9 +55,16 @@ export const MapSelectorContainer: FunctionComponent<IMapSelectorContainerProps>
...
+        if (!pid && !pin) {
           return property;
         }
+        const queryObject = {};
+        if (pid.length > 0) {
+          queryObject['PID'] = pid;
+        }
+        if (isValidId(+pin)) {
+          queryObject['PIN'] = pin;
+        }
```
Comment (asanchezr) [inline source/frontend/src/components/propertySelector/MapSelectorContainer.tsx:67]: Could you please comment what we want to do with this **queryObject**? It seems it gets created but not used anywhere. Perhaps it is meant to be used in the line below (line 68)?

**Comment by asanchezr** [comment]
@devinleighsmith - please update snapshots
image

**Pull request merged**